plugins/woocommerce-blocks/assets/js/base/components/cart-checkout/place-order-button/index.tsx

```diff
@@ -23,14 +33,46 @@ const PlaceOrderButton = ( {
 		waitingForRedirect,
 	} = useCheckoutSubmit();
 
+	const { cartTotals } = useStoreCart();
+	const totalsCurrency = getCurrencyFromPriceResponse( cartTotals );
+
+	const buttonLabel = (
+		<div
+			// Hide this from screen readers while the checkout is processing. The text will not be removed from the
+			// DOM, it will just be hidden with CSS to maintain the button's size while the spinner appears.
+			aria-hidden={ waitingForProcessing || waitingForRedirect }
+			className={ clsx(
+				'wc-block-components-checkout-place-order-button__text',
+				{
+					'wc-block-components-checkout-place-order-button__text--visually-hidden':
+						waitingForProcessing || waitingForRedirect,
+				}
+			) }
+		>
+			{ label }
+			{ showPrice && (
+				<div className="wc-block-components-checkout-place-order-button__separator" />
+			) }
+			{ showPrice && (
+				<div className="wc-block-components-checkout-place-order-button__price">
+					<FormattedMonetaryAmount
+						value={ cartTotals.total_price }
+						currency={ totalsCurrency }
+					/>
+				</div>
+			) }
```

Comment:
Why aren't those 2 in the same condition under an empty fragment?

Reply:
Then they get output in the same div and mess up the flex of the parent.

Reply:
A fragment won't make it to the DOM
```suggestion
			{ showPrice && (
				<>
					<div className="wc-block-components-checkout-place-order-button__separator" />
					<div className="wc-block-components-checkout-place-order-button__price">
						<FormattedMonetaryAmount
							value={ cartTotals.total_price }
							currency={ totalsCurrency }
						/>
					</div>
				</>
			) }
```